Source/Parallelization/WarpXRegrid.cpp

```diff
@@ -166,6 +166,61 @@ WarpX::RemakeLevel (int lev, Real /*time*/, const BoxArray& ba, const Distributi
             rho_fp[lev] = std::move(pmf);
         }
 
+#ifdef WARPX_USE_PSATD
+        if (maxwell_solver_id == MaxwellSolverAlgo::PSATD) {
+            if (spectral_solver_fp[lev] != nullptr) {
+                // Get the cell-centered box
+                BoxArray realspace_ba = ba;   // Copy box
+                realspace_ba.enclosedCells(); // Make it cell-centered
+                auto ngE = getngE();
+                auto dx = CellSize(lev);
+#   if (AMREX_SPACEDIM == 3)
+                RealVect dx_vect(dx[0], dx[1], dx[2]);
+#   elif (AMREX_SPACEDIM == 2)
+                RealVect dx_vect(dx[0], dx[2]);
+#   endif
+
+#   ifdef WARPX_DIM_RZ
+                if ( fft_periodic_single_box == false ) {
+                    realspace_ba.grow(1, ngE[1]); // add guard cells only in z
+                }
+                AllocLevelSpectralSolverRZ(spectral_solver_fp,
+                                           lev,
+                                           realspace_ba,
+                                           dm,
+                                           n_rz_azimuthal_modes,
```

Comment:
Many of these variables in the argument list are not necessary since they are WarpX attributes and can be referred to directly without being passed in.

Reply:
Ah I see, I have removed the arguments we can refer to directly.  How is this looking now?